src/functions.php

```diff
@@ -51,6 +51,10 @@ function server($name, $domain = null, $port = 22)
     $deployer->servers->set($name, $server);
     $deployer->environments->set($name, $env);
 
+    // Setting environment variables about the server automatically
+    $env->set('name', $server);
+    $env->set('host', $host);
```

Comment:
I think it's better to put this lines inside `Builder` class, as there are `localServer` function needs this lines too.

Also can you add some test to check after function `server`/`localServer` call env vars a created?